src/ir/datalog/program.h

```diff
@@ -127,6 +137,12 @@ class RelationDeclaration {
   absl::string_view relation_name() const { return relation_name_; }
   bool is_attribute() const { return is_attribute_; }
   const std::vector<Argument>& arguments() const { return arguments_; }
+  
+  bool operator==(const RelationDeclaration& otherDeclaration) const {
+    return this->relation_name_ == otherDeclaration.relation_name() &&
+    this->is_attribute_ == otherDeclaration.is_attribute() &&
+    this->arguments_ == otherDeclaration.arguments();
```

Comment:
Use the fields directly?
```suggestion
    return this->relation_name_ == otherDeclaration.relation_name_ &&
    this->is_attribute_ == otherDeclaration.is_attribute_ &&
    this->arguments_ == otherDeclaration.arguments_;
```

Reply:
fixed